### ISSUE TITLE
TargetRubyVersion is loaded from a higher-level when getting file path exclusions

### ISSUE THREAD

**Issue by pirj** [rubocop-hq/rubocop#7911]
RuboCop [is adding `AllCops/Exclude`](https://github.com/rubocop-hq/rubocop/blob/172b8a0dc9be293ada1c97df6812238883e3bf13/lib/rubocop/config_loader.rb#L117) from **the highest** up the file hierarchy `.rubocop.yml` it can find.
`AllCops/TargetRubyVersion` from e.g. `~/.rubocop.yml` is never used, [only `AllCops/Exclude`](https://github.com/rubocop-hq/rubocop/blob/172b8a0dc9be293ada1c97df6812238883e3bf13/lib/rubocop/config.rb#L92).

However, if RuboCop is of a recent version with dropped `TargetRubyVersion` support, it fails the validation.

`rubocop --ignore_parent_exclusion` passes just fine.

It makes sense to skip Ruby version validation (and maybe cops validations) when those config files are loaded only to get file path exclusions from it.

Side observation: It's strange that exclusions are only taken from **the highest** up the hierarchy, not gathered from all of them. It's a matter of design though.

Originally reported by @firedev [here](https://github.com/rubocop-hq/rubocop-rspec/issues/898).

## Expected behavior

RuboCop proceeds: no `TargetRubyVersion` validation is performed on higher-level `.rubocop.yml` files.

## Actual behavior

RuboCop fails on `TargetRubyVersion` validation that is performed when loading higher-level `.rubocop.yml` file when getting `AllCops/Exclude` from it:

```
$ rubocop
Error: RuboCop found unsupported Ruby version 2.3

$ rubocop --ignore_parent_exclusion
Inspecting xxx files

~/Sites/work $ rubocop --debug
For /Users/user/Sites/work: configuration from /Users/user/Sites/work/.rubocop.yml
configuration from /Users/user/.rbenv/versions/2.6.6/lib/ruby/gems/2.6.0/gems/rubocop-rspec-1.38.1/config/default.yml
configuration from /Users/user/.rbenv/versions/2.6.6/lib/ruby/gems/2.6.0/gems/rubocop-rspec-1.38.1/config/default.yml
Default configuration from /Users/user/.rbenv/versions/2.6.6/lib/ruby/gems/2.6.0/gems/rubocop-0.82.0/config/default.yml
/Users/user/Sites/work/.rubocop.yml: Warning: no department given for Documentation.
Warning: Layout/ClosingParenthesisIndentation does not support EnforcedStyle parameter.

Supported parameters are:

  - Enabled

AllCops/Exclude configuration from /Users/user/.rubocop.yml
/Users/user/.rubocop.yml: Warning: no department given for Documentation.
/Users/user/.rubocop.yml: Warning: no department given for AsciiComments.
/Users/user/.rubocop.yml: Warning: no department given for DotPosition.
Error: RuboCop found unsupported Ruby version 2.3 in `TargetRubyVersion` parameter (in /Users/user/.rubocop.yml). 2.3-compatible analysis was dropped after version 0.82.
Supported versions: 2.4, 2.5, 2.6, 2.7
Finished in 0.24214499999652617 seconds
```

## Steps to reproduce the problem

Create a `~/.rubocop.yml` file with the following contents:
```yml

AllCops:
  TargetRubyVersion: 2.3
  Exclude: docs
```

## RuboCop version

```
$ [bundle exec] rubocop -V
0.82.0
```
